### PR TITLE
Payload verification

### PR DESCRIPTION
**Pull request by ryanblock**

### CODE REVIEW

File: src/http/post-preview-000u-000r/_verify.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    const { u: user, r: repo } = req.pathParameters;
+    const { pr, sha } = req.body;
+
+    const base = 'https://api.github.com';
+    const url = `${base}/repos/${user}/${repo}/pulls/${pr}/commits`;
+
+    const commits = await get({ url });
+
+    // Is this an actual SHA on the repo?
+    const shaIsReal = commits.body.find(c => c.sha === sha);
```
Comment (ljharb) [inline src/http/post-preview-000u-000r/_verify.js:21]:
```suggestion
    const shaIsReal = commits.body.some(c => c.sha === sha);
```

File: src/http/post-preview-000u-000r/_verify.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    // Have we seen this SHA before and published its preview?
+    const shaHasBeenSeen = await data.get({
```
Comment (ljharb) [inline src/http/post-preview-000u-000r/_verify.js:24]: is this about the same as using `count`?

Comment (ryanblock) [reply src/http/post-preview-000u-000r/_verify.js:24]: Count is a query, which assumes multiple results would be feasible – the key on this is the sha, so it should be individual.

Comment (ljharb) [reply src/http/post-preview-000u-000r/_verify.js:24]: mainly just asking if "count is 0" was more efficient than "get the entire data and ignore it"

**Pull request merged**